txnsync/bitmask.go

```diff
@@ -142,67 +148,75 @@ func (b *bitmask) iterate(entries int, maxIndex int, callback func(int, int) err
 		for i, v := range (*b)[1:] {
 			for j := 0; j < 8 && v < 255; j++ {
 				if v&1 == 0 {
-					if index >= maxIndex {
+					if setBitCounter >= countOfSetBits {
 						return errDataMissing
 					}
-					if err := callback(8*i+j, index); err != nil {
+					entryIndex := 8*i + j
+					if entryIndex >= countOfEntries {
+						return errIndexNotFound
+					}
+					if err := callback(entryIndex, setBitCounter); err != nil {
 						return err
 					}
-					index++
+					setBitCounter++
 				}
 				v >>= 1
 			}
 		}
-		for i := (len(*b) - 1) * 8; i < entries; i++ {
-			if index >= maxIndex {
+		for i := (len(*b) - 1) * 8; i < countOfEntries; i++ {
+			if setBitCounter >= countOfSetBits {
 				return errDataMissing
 			}
-			if err := callback(i, index); err != nil {
+			if err := callback(i, setBitCounter); err != nil {
 				return err
 			}
-			index++
+			setBitCounter++
 		}
 	case 2:
-		sum := 0
+		sum := 0 // entryIndex
```

Comment:
we can just use the desired variable name here instead of sum, and same for case 3

Reply:
maybe not for case 3